src/protocols/aeternity/composables/aeNames.ts

```diff
@@ -0,0 +1,290 @@
+
+    return preferred.value[`${address}-${nodeNetworkId.value}`] || '';
+  }
+
+  function getAuction(name: string): IAuction {
+    return auctions.value[name] || null;
+  }
+
+  function getHighestBid(name: string): IAuctionBid {
+    return getAuction(name)?.bids.reduce((a, b) => (a.nameFee.isGreaterThan(b.nameFee) ? a : b));
+  }
+
+  async function fetchOwnedNames() {
+    areNamesFetching.value = true;
+
+    function fetchPendingNameClaimTransactions(address: Encoded.AccountAddress) {
+      return fetchPendingTransactions(address)
+        .then(
+          (transactions: ITransaction[]) => (transactions)
+            .filter(({ tx: { type } }) => type === 'NameClaimTx')
+            .map(({ tx, ...otherTx }) => ({
+              ...otherTx,
+              ...tx,
+              owner: tx.accountId,
+            })),
+        );
+    }
+
+    const middleware = await getMiddleware();
+
+    ownedNames.value = await Promise.all(
+      aeAccounts.value.map(({ address }) => Promise.all([
+        fetchPendingNameClaimTransactions(address),
+        fetchAllPages(
+          () => middleware.getNames({ owned_by: address, state: 'active', limit: 100 }),
+          fetchFromMiddlewareCamelCased,
+        ).then((data) => data.map(({ info, name, hash }) => ({
+          createdAtHeight: info.activeFrom,
+          expiresAt: info.expireHeight,
+          owner: info.ownership.current,
+          pointers: info.pointers,
+          autoExtend: (
+            ownedNames.value.find((n) => n.name === name)?.autoExtend
+            || pendingAutoExtendNames.value?.includes(name)
+          ),
+          name,
+          hash,
+        }))),
+      ])),
+    ).then((arr) => arr.flat(2)) as IName[];
+
+    areNamesFetching.value = true;
+  }
+
+  async function updatePointer(
+    {
+      name,
+      address,
+      type = UPDATE_POINTER_ACTION.update,
+    }: IUpdatePointerParams,
+  ) {
+    const aeSdk = await getAeSdk();
+    const nameEntry = await aeSdk.aensQuery(name);
+    try {
+      if (type === UPDATE_POINTER_ACTION.extend) {
+        // @ts-ignore
+        await nameEntry.extendTtl();
+      } else if (type === UPDATE_POINTER_ACTION.update) {
+        await aeSdk.aensUpdate(name, { account_pubkey: address }, { extendPointers: true });
+      }
+      openDefaultModal({
+        msg: tg('pages.names.pointer-added', { type }),
+      });
+    } catch (error: any) {
+      if (error.message.includes('Account not found')) {
+        handleUnknownError(error);
+      } else {
+        openDefaultModal({
+          msg: isInsufficientBalanceError(error)
+            ? tg('modals.insufficient-balance.msg')
+            : error.message,
+        });
+      }
+    }
+  }
+
+  function setDefault({ address, name }: IAddressNamePair) {
+    if (name) {
+      defaults.value[`${address}-${nodeNetworkId.value}`] = name;
+    } else {
+      delete defaults.value[`${address}-${nodeNetworkId.value}`];
+    }
+  }
+
+  function setDefaultNames() {
```

Comment:
As this is not a setter I suggest to follow same pattern as other libs/composables and rename this to `updateDefaultNames`.